File: pset6/sentimental-credit/credit.py
# TODO
import re

# Loop to check each card through regex
while True:
    # get input from user
    cc = input("Credit Card Number: ")
    # conditionals to check each card
    if re.match("^4[0-9]{12}(?:[0-9]{3})?$", cc):
        print("VISA\n")
        break
    elif re.match("^(?:5[1-5][0-9]{2}|222[1-9]|22[3-9][0-9]|2[3-6][0-9]{2}|27[01][0-9]|2720)[0-9]{12}$", cc):
        print("MASTERCARD\n")
        break
    elif re.match("^3[47][0-9]{13}$", cc):
        print("AMEX\n")
        break
    else:
        print("INVALID\n")
        break

File: pset1/credit/credit.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    //prompt for credit card number
    long cc;
    do
    {
        cc = get_long("Number: \n");
    }
    while (cc < 0);

    long ccs = cc;
    long ccd = cc;
    long ldc = cc;
    int lda = 0;
    int count;
    int sume = 0;
    for (count = 0, lda = 0; ccs > 0; count++)
    {
        lda = ccs % 10;
        ccs = ccs / 10;
        lda = lda * 2;
        if (lda >= 10)
        {
            lda = lda % 10 + lda / 10;
        }
        else
            ;
        if (count % 2 == 0)
            ;
        else
        {
            sume = sume + lda;
        }
    }
    int ldb;
    int counta;
    int sumo = 0;
    for (counta = 0, ldb = 0; ccd > 0; counta++)
    {
        ldb = ccd % 10;
        ccd = ccd / 10;
        if (counta % 2 == 0)
        {
            sumo = sumo + ldb;
        }
        else
            ;
    }
    int amex = 0;
    int master = 0;
    int countb;
    int countc;
    for (countb = 0; ldc >= 100 ; countb++)
    {
        ldc = ldc / 10;
    }

    amex = ldc;
    master = ldc;
    int visa = ldb;
    int val = sumo + sume;
    if (val % 10 == 0 && (count == 13 || count == 16) && visa == 4)
    {
        printf("VISA\n");
    }
    else if (val % 10 == 0 && count == 15 && (amex == 34 || amex == 37))
    {
        printf("AMEX\n");
    }
    else if (val % 10 == 0 && count == 16 && (master == 51 || master == 52 || master == 53 || master == 54 || master == 55))
    {
        printf("MASTERCARD\n");
    }
    else
    {
        printf("INVALID\n");
    }
}